test/test_script.py

```diff
@@ -1174,6 +1175,42 @@ def test_cmd_datetime(self):
             # Restore original datetime object
             datetime.datetime = original_datetime
 
+    def test_cmd_datetime_platform_dependent(self):
+        # Platform dependent testing because different platforms (both os and Python version)
+        # support some format arguments differently.
+        possible_tests = (
+            '%',    # Hanging % at end of format
+            '%-d',  # Non zero-padded day
+            '%-m',  # Non zero-padded month
+            '%3Y',  # Length specifier shorter than string
+        )
+        tests_to_run = []
+        # Get list of tests for unsupported format codes
+        for test_case in possible_tests:
+            try:
+                datetime.datetime.now().strftime(test_case)
+            except ValueError:
+                tests_to_run.append(test_case)
+        if not tests_to_run:
+            self.skipTest('datetime module supports all test cases')
+        # Save original datetime object and substitute one returning
+        # a fixed now() value for testing.
+        original_datetime = datetime.datetime
+        datetime.datetime = _DateTime
+
+        try:
+            context = Metadata()
+            areg = r"^\$datetime:1:\d+: Unsupported format code"
+            # Tests with invalid format code (platform dependent tests)
+            for test_case in tests_to_run:
+                with self.assertRaisesRegex(ScriptRuntimeError, areg):
+                    self.parser.eval(r'$datetime(\{0})'.format(test_case))
+        except (AssertionError, ValueError, IndexError) as err:
+            raise err
```

Comment:
@phw @rdswift I'm not sure this `except ... as err` / `raise err` is needed. `try` / `finally` might be enough

Reply:
Of course it's not necessary.  In fact, it's dumb the way I wrote it.  What was I thinking?  :blush: